### ISSUE TITLE
1.3.1 breaks sub-error

### ISSUE THREAD

**Issue by unional** [JsCommunity/make-error#21]
1.3 is failing this test: https://github.com/blakeembrey/make-error-cause/blob/master/src/index.spec.ts#L19-L20

while 1.2 is working fine.

```sh
      expected: |-
        'SubTestError: more bad\nCaused by: TestError: something bad\nCaused by: Error: boom!'
      actual: |-
        'TestError: more bad\nCaused by: TestError: something bad\nCaused by: Error: boom!'
      at: Test.<anonymous> (/Users/hwong/code/unional/make-error-cause/src/index.spec.ts:20:7)
      stack: |-
```

**Comment by unional** [comment]
The break is introduced in 1.3.1. 1.3.0 is working.

**Comment by unional** [comment]
Here is the failing test case:

```js
  it('sub error should have its own name', function () {
    var TestError = makeError('TestError')
    var SubTestError = makeError('SubTestError', TestError)

    var again = new SubTestError('more bad')
    expect(again.name).toBe('SubTestError') // `again.name` is `TestError`
  })
```

The problem comes from the usage or `Reflect.construct()`

**Comment by unional** [comment]
cc/@blakeembrey

**Comment by julien-f** [comment]
Thank you for your report, fix has been released in 1.3.4.

I've made a major refactoring of the tests and there are still a number of issues to resolve, I'm afraid it will not be easy and might require a change in the way we do inheritance with ES5.
